plans/terraform/create.pp

```diff
@@ -0,0 +1,108 @@
+#
+# @summary Plan to provision a Virtual Machine using Terraform for different cloud providers (GCP, AWS & Azure)
+#
+# @param tf_dir
+#   The directory where the Terraform module is located
+#
+# @param resource_name
+#   The name of the resource to be provisioned
+#
+# @param instance_type
+#   The instance type to be provisioned
+#
+# @param image
+#   The cloud image that is used for new instance provisioning, format differs
+#   depending on provider
+#
+# @param region
+#   Which region to provision infrastructure in, if not provided default will
+#   be determined by provider
+#
+# @param ssh_key
+#   The SSH key name to be provisioned & access the machine
+#
+# @param ssh_user
+#   The ssh user to provision on provisioned VMs
+#
+# @param subnet
+#   Name or ID of an existing subnet to attach newly provisioned VMs
+#
+# @param security_group_ids
+#   The list of security group which will be attached to the newly provisioned VMs
+#
+# @param backend
+#   The backend to manage the state of the Terraform project
+#
+# @param bucket
+#   The bucket where tfstate will get maintained
+#
+# @param profile
+#   The name of the profile to be used for provisioning
+#
+# @param tags
+#  The list of tags to be attached to the newly provisioned VMs
+#
+plan provision::terraform::create(
+  String[1] $tf_dir                    = undef,
+  Enum['gcp', 'aws', 'azure'] $provider = 'aws',
+  Enum['xlarge', 'large', 'medium', 'small', 'micro'] $instance_type = 'micro',
```

Comment:
Do make this DRY you could create a type definition, see for example https://github.com/puppetlabs/puppetlabs-peadm/blob/main/types/singletargetspec.pp